internal/webhooks/runtime/extensionconfig_webhook.go

```diff
@@ -93,10 +101,115 @@ func (webhook *ExtensionConfig) validate(_ context.Context, _, _ *runtimev1.Exte
 			"can be set only if the RuntimeSDK feature flag is enabled",
 		)
 	}
+
+	var allErrs field.ErrorList
+	allErrs = append(allErrs, validateExtensionConfigSpec(newExtensionConfig)...)
+
+	if len(allErrs) > 0 {
+		return apierrors.NewInvalid(runtimev1.GroupVersion.WithKind("ExtensionConfig").GroupKind(), newExtensionConfig.Name, allErrs)
+	}
 	return nil
 }
 
 // ValidateDelete implements webhook.Validator so a webhook will be registered for the type.
 func (webhook *ExtensionConfig) ValidateDelete(_ context.Context, _ runtime.Object) error {
 	return nil
 }
+
+func validateExtensionConfigSpec(e *runtimev1.ExtensionConfig) field.ErrorList {
+	var allErrs field.ErrorList
+
+	specPath := field.NewPath("spec")
+
+	if e.Spec.ClientConfig.URL == nil && e.Spec.ClientConfig.Service == nil {
+		allErrs = append(allErrs, field.Required(
+			specPath.Child("clientConfig"),
+			"either URL or Service must be defined",
+		))
+	}
+	if e.Spec.ClientConfig.URL != nil && e.Spec.ClientConfig.Service != nil {
+		allErrs = append(allErrs, field.Forbidden(
+			specPath.Child("clientConfig"),
+			"only one of URL or Service can be defined",
+		))
+	}
+
+	// Validate URl
+	if e.Spec.ClientConfig.URL != nil {
+		uri, err := url.ParseRequestURI(*e.Spec.ClientConfig.URL)
+		if err != nil {
+			allErrs = append(allErrs, field.Invalid(
+				specPath.Child("clientConfig", "url"),
+				*e.Spec.ClientConfig.URL,
+				"must be a valid URL, e.g. https://example.com",
+			))
+		}
+		if uri.Scheme != "http" && uri.Scheme != "https" {
```

Comment:
I removed the `// TODO: Decide handling of http/https prefixes.` here and added an additional check for the scheme.

From previous discussions with @sbueringer : It makes sense to keep `http` especially for development perspective?

Mutating or Validatingwebhooks enforce `https`.

Reply:
moved to updated code